modules/post/windows/gather/forensics/fanny_bmp_check.rb

```diff
@@ -0,0 +1,84 @@
+# Detect fanny.bmp artifacts in registry
+# William M. william-martens@protonmail.ch
+# 
+
+
+require 'msf/core'
+require 'msf/core/post/common'
+require 'msf/core/post/windows/priv'
+class MetasploitModule < Msf::Post
+
+        include Msf::Post::Common
+        include Msf::Post::Windows::Registry
+
+
+        def initialize(info={})
+                super( update_info( info,
+                                 'Name' => 'FannyBMP Registry Check',
+                                 'Description' => %q{ This module searches for the Fanny.bmp worm - related registry keys},
+                                 'License' => MSF_LICENSE,
+                                 'Author' => [ 'William M.'],
+                                 'Version' => '$Revision$', 
+                                 'Platform' => [ 'windows' ],
+                                 'SessionTypes' => [ 'meterpreter' ],
+                                 'References' => [[ 'URL', 'https://securelist.com/a-fanny-equation-i-am-your-father-stuxnet/68787' ]]
+                        ))
+
+        end
+        
+        def run
+                                        ##################  artifacts  ######################################
+                                        # From:  https://securelist.com/a-fanny-equation-i-am-your-father-stuxnet/68787
+                                		    # HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4
+                                				# HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4\Driver
+                                				# HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4\filter2
+                                				# HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4\filter3
+                                				# HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4\filter8
+                                        ##################  artifacts  ######################################
+                                                            
+                                query =    'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\"ECELP4",'
+                                query += 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4\Driver,'
+                                query += 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4\filter2,'
+                                query += 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4\filter3,'
+                                query += 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\MediaResources\acm\ECELP4\filter8'
+                                
+                match = 0
+
+                                print_status("Searching registry on #{sysinfo['Computer']} for Fanny.bmp artifacts.")
+                keys = query.split(/,/)
```

Comment:
Rather than define these as a string then split, it would make more sense to define these as an array/